compiler/passes/convert-uast.cpp

```diff
@@ -1401,6 +1401,10 @@ struct Converter {
       } else if (const uast::ReduceIntent* rd = expr->toReduceIntent()) {
         astlocMarker markAstLoc(rd->id());
 
+        if(fForeachIntents && parent->toForeach()) {
```

Comment:
Using `fForeachIntents` check here feels redundant, but I am still reading code.

Reply:
Yeah looks unnecessary.

I was a bit worried we might hit that error message before we check for the `with` clause more generally but I just tried it out and we don't. I'll simplify to just `if(parent->toForeach()) { ... }`